cpp/daal/include/data_management/data/row_merged_numeric_table.h

```diff
@@ -117,6 +118,38 @@ class DAAL_EXPORT RowMergedNumericTable : public NumericTable
         return setNumberOfRowsImpl(_obsnum + obs);
     }
 
+    /**
+     * Converts the table to the single SyclHomogenNumericTable
+     * \param[out] stat  Status of the conversion
+     */
+    template <typename T>
+    NumericTablePtr toSyclHomogen(services::Status & stat)
```

Comment:
We don't expected to mix host-only functionality and SYCL in old interfaces. I recommend to move this as free function to separate header file in `data_management/data/internal` like `numeric_table_sycl_utils.h`.

Don't forget to add `DAAL_SYCL_INTERFACE` ifdef.

Reply:
Reworked to be similar with ConvertToHomogen